backend/site-api/src/app/site/update_site.rs

```diff
@@ -64,6 +64,15 @@ pub async fn update_site(
             DbError::NoUpdate => ApiError::bad_request()
                 .code(ApiErrorCode::NoUpdates)
                 .message("No updates were made for the request.".to_string()),
+            DbError::EntityNotFound() => {
+                if has_update_key {
+                    ApiError::bad_request()
+                        .code(ApiErrorCode::UpdateStale)
```

Comment:
I added an error code here, I think it will make it easier to implement the frontend part.